lib/autobuild/importer.rb

```diff
@@ -336,7 +336,8 @@ def perform_update(package,only_local=false)
             package.message "update failed in #{package.importdir}, retrying (#{retry_count}/#{self.retry_count})"
             retry
         ensure
-            package.progress_done "updated %s"
+            update_msg = did_update ? "updated" : "already up-to-date"
```

Comment:
I would have checked against 'true' and 'false' explicitely. This way, if some codepaths have not been updated, one does not get a (possibly confusing) "already-up-to-date".

Moreover, style-wise, I avoid `?:` in Ruby (well, actually, in C++ too). I prefer to use an explicit if else

~~~
update_msg = if did_update == false then 'already up-to-date'
                       else 'updated'
                       end
~~~

Reply:
> This way, if some codepaths have not been updated, one does not get a (possibly confusing) "already-up-to-date"

Good point..

> Moreover, style-wise, I avoid ?: in Ruby (well, actually, in C++ too)

I will accept that for ruby since I'm still getting used to it but in C++ the ternary operator is my bread-and-butter for conciseness. Why would you avoid that?

Reply:
I just find it hard to read ... Moreover, in Ruby, since the `if ... else ... end` is an expression you can use it as a more verbose `?:`